openjdk/mmtkUpcalls.cpp

```diff
@@ -49,6 +49,12 @@ static void mmtk_stop_all_mutators(void *tls, void (*create_stack_scan_work)(voi
   MMTkHeap::heap()->companion_thread()->request(MMTkVMCompanionThread::_threads_suspended, true);
   log_debug(gc)("Mutators stopped. Now enumerate threads for scanning...");
 
+  ClassLoaderDataGraph::clear_claimed_marks();
```

Comment:
These added lines of code happen after all threads stopped, and before we create any work packets for stack scanning.  Is this necessary?

If we allow more "eager" stack scanning, i.e. create one work packet for each thread when the thread is visited during the `companion_thread()->request()` above, these lines seem to have no place to fit in.

Reply:
Fixed. This code is moved before making the `_threads_suspended` request.